sonic-utilities-data/templates/sonic-cli-gen/config.py.j2

```diff
@@ -0,0 +1,559 @@
+        table (str): Table name to add new entry to.
+        key (Union[str, Tuple]): Key name in the table.
+        data (Dict): Entry data.
+    Raises:
+        Exception: when cfg does not satisfy YANG schema.
+    """
+
+    cfg = db.get_config()
+    cfg.setdefault(table, {})
+    if key in cfg[table]:
+        raise Exception(f"{key} already exists")
+
+    cfg[table][key] = data
+
+    validate_config_or_raise(cfg)
+    db.set_entry(table, key, data)
+
+
+def update_entry_validated(db, table, key, data, create_if_not_exists=False):
+    """ Update entry in table and validate configuration.
+    If attribute value in data is None, the attribute is deleted.
+
+    Args:
+        db (swsscommon.ConfigDBConnector): Config DB connector obect.
+        table (str): Table name to add new entry to.
+        key (Union[str, Tuple]): Key name in the table.
+        data (Dict): Entry data.
+        create_if_not_exists (bool):
+            In case entry does not exists already a new entry
+            is not created if this flag is set to False and
+            creates a new entry if flag is set to True.
+    Raises:
+        Exception: when cfg does not satisfy YANG schema.
+    """
+
+    cfg = db.get_config()
+    cfg.setdefault(table, {})
+
+    if create_if_not_exists:
+        cfg[table].setdefault(key, {})
+
+    if key not in cfg[table]:
+        raise Exception(f"{key} does not exist")
+
+    for attr, value in data.items():
+        if value is None and attr in cfg[table][key]:
+            cfg[table][key].pop(attr)
+        else:
+            cfg[table][key][attr] = value
+
+    validate_config_or_raise(cfg)
+    db.set_entry(table, key, cfg[table][key])
+
+
+def del_entry_validated(db, table, key):
+    """ Delete entry in table and validate configuration.
+
+    Args:
+        db (swsscommon.ConfigDBConnector): Config DB connector obect.
+        table (str): Table name to add new entry to.
+        key (Union[str, Tuple]): Key name in the table.
+    Raises:
+        Exception: when cfg does not satisfy YANG schema.
+    """
+
+    cfg = db.get_config()
+    cfg.setdefault(table, {})
+    if key not in cfg[table]:
+        raise Exception(f"{key} does not exist")
+
+    cfg[table].pop(key)
+
+    validate_config_or_raise(cfg)
+    db.set_entry(table, key, None)
+
+
+def add_list_entry_validated(db, table, key, attr, data):
+    """ Add new entry into list in table and validate configuration.
+
+    Args:
+        db (swsscommon.ConfigDBConnector): Config DB connector obect.
+        table (str): Table name to add data to.
+        key (Union[str, Tuple]): Key name in the table.
+        attr (str): Attribute name which represents a list the data needs to be added to.
+        data (List): Data list to add to config DB.
+    Raises:
+        Exception: when cfg does not satisfy YANG schema.
+    """
+
+    cfg = db.get_config()
+    cfg.setdefault(table, {})
+    if key not in cfg[table]:
```

Comment:
Can't  list entry be the only entry for the key ?
In other words, why this call should not create the key, if not exists ?

Reply:
"config 'object' 'list' add" should fail if no object is created in config db. It is same operation as "update" which fails if no key to update is present.